### PR TITLE
tweak the executor implementation to better match origin behavior

### PR DESCRIPTION
**Pull request by panyx0718**
test=develop

### CODE REVIEW

File: python/paddle/fluid/executor.py [hunk 1]
```diff
@@ -589,11 +588,11 @@ def _run(self, program, feed, fetch_list, feed_var_name, fetch_var_name,
...
+        exe.run(program.desc, scope, 0, True, True)
         outs = self._fetch_data(fetch_list, fetch_var_name, scope)
         if return_numpy:
             outs = as_numpy(outs)
         return outs
 
-    def _run_inference(self, program, feed):
-        return self.executor.run(feed)
+    def _run_inference(self, exe, feed):
+        return exe.run(feed)
```
Comment (chengduoZH) [inline python/paddle/fluid/executor.py:598]: Doesn't `Inference` need fetch data?

File: python/paddle/fluid/executor.py [hunk 2]
```diff
@@ -397,12 +399,13 @@ def close(self):
...
         """
-        if not self._closed and self.executor:
-            self.executor.close()
+        if not self._closed:
+            self._default_executor.close()
             self._closed = True
 
     def _run_parallel(self, program, scope, feed, fetch_list, fetch_var_name,
                       return_numpy):
+        exe = program._executor
```
Comment (chengduoZH) [inline python/paddle/fluid/executor.py:408]: `program` is a bit confusing with `fluid.Program`, it should be `compiled_program`.
And the `exe` should be `parallel_exe`.

File: python/paddle/fluid/executor.py [hunk 3]
```diff
@@ -414,8 +417,7 @@ def _run_parallel(self, program, scope, feed, fetch_list, fetch_var_name,
                     feed_tensor.set(feed[feed_name], core.CPUPlace())
                 feed_tensor_dict[feed_name] = feed_tensor
 
-            self.executor.feed_and_split_tensor_into_local_scopes(
-                feed_tensor_dict)
+            exe.feed_and_split_tensor_into_local_scopes(feed_tensor_dict)
         elif isinstance(feed, list) or isinstance(feed, tuple):
             if len(feed) != len(program._places):
```
Comment (chengduoZH) [inline python/paddle/fluid/executor.py:422]: Same above, program is a bit confusing with fluid.Program

Comment (panyx0718) [reply python/paddle/fluid/executor.py:422]: not a big problem now, can improve in the future.

File: python/paddle/fluid/executor.py (hunk 2 above)
Comment (panyx0718) [reply python/paddle/fluid/executor.py:408]: not a big problem now, can improve in the future.

File: python/paddle/fluid/executor.py (hunk 1 above)
Comment (panyx0718) [reply python/paddle/fluid/executor.py:598]: This is inference's problem, should improve in the future.

**Review by chengduoZH — APPROVED** [review]

**Pull request merged**